python/packages/autogen-core/src/autogen_core/tools/_base.py

```diff
@@ -148,13 +148,10 @@ def return_value_as_string(self, value: Any) -> str:
     async def run(self, args: ArgsT, cancellation_token: CancellationToken) -> ReturnT: ...
 
     async def run_json(self, args: Mapping[str, Any], cancellation_token: CancellationToken) -> Any:
-        with get_tracer("base_tool").start_as_current_span(
-            self._name,
-            attributes={
-                "tool_name": self._name,
-                "tool_description": self._description,
-                "tool_args": json.dumps(args),
-            },
+        with trace_tool_span(
```

Comment:
The default impl here doesn't pass in the `tool_call_id` which is probably useful for correlating maintaining deeper historical context. It seems annoying to add that to the function definition, but could be good to add somehow.